dspMetadataGUI/collectMetadata.py

```diff
@@ -0,0 +1,1295 @@
+        self.metadataset = metadataset
+        self.rows = []
+        self.multiple_selection = 0
+        outer_sizer = wx.BoxSizer(wx.VERTICAL)
+        sizer = wx.FlexGridSizer(4, 10, 10)
+
+        if dataset:
+            ds = dataset
+            if multiple:
+                ds = self.active_dataset
+            for i, prop in enumerate(ds.get_properties()):
+                row = PropertyRow(self, prop, sizer, i, metadataset)
+                self.rows.append(row)
+
+        sizer.AddGrowableCol(1)
+
+        if multiple:
+            dataset_sizer = wx.BoxSizer()
+            dataset_listbox = wx.ListBox(self, size=(200, -1))
+            for ds in dataset:
+                dataset_listbox.Append(str(ds))
+            dataset_listbox.Bind(
+                wx.EVT_LISTBOX, lambda e: self.change_selection(e))
+            dataset_listbox.Select(0)
+            self.multiple_listbox = dataset_listbox
+            dataset_sizer.Add(dataset_listbox, wx.EXPAND)
+            dataset_sizer.AddSpacer(5)
+            button_sizer = wx.BoxSizer(wx.VERTICAL)
+            button_add = wx.Button(self, label="Add New")
+            button_add.Bind(wx.EVT_BUTTON, lambda e: self.add_object(
+                e, dataset_listbox, title))
+            button_sizer.Add(button_add, flag=wx.EXPAND)
+            button_remove = wx.Button(self, label="Remove Selected")
+            button_remove.Bind(
+                wx.EVT_BUTTON, lambda e: self.remove_object(e, dataset_listbox))
+            button_sizer.Add(button_remove)
+            dataset_sizer.Add(button_sizer)
+            outer_sizer.Add(dataset_sizer, flag=wx.EXPAND)
+            outer_sizer.AddSpacer(20)
+        outer_sizer.Add(sizer, flag=wx.EXPAND)
+        self.SetSizer(outer_sizer)
+        self.SetupScrolling(scroll_x=False, scroll_y=True)
+        self.Fit()
+        self.Layout()
+        # self.SetScrollbars(0, 16, 60, 15)
+
+    @property
+    def active_dataset(self):
+        if self.multiple:
+            return self.dataset[self.multiple_selection]
+        else:
+            return self.dataset
+
+    def update_data(self):
+        for row in self.rows:
+            row.update_data()
+
+    def refresh_ui(self):
+        if self.multiple:
+            self.multiple_listbox.SetItems([str(ds) for ds in self.dataset])
+            if self.multiple_selection < 0:
+                self.multiple_selection = len(self.multiple_listbox.GetItems()) - 1
+            self.multiple_listbox.SetSelection(self.multiple_selection)
+        for row in self.rows:
+            row.refresh_ui()
+        self.Layout()
+
+    def add_object(self, event, listbox, title):
+        if title == "Person":
+            self.metadataset.add_person()
+        if title == "Dataset":
+            self.metadataset.add_dataset()
+        elif title == "Organization":
+            self.metadataset.add_organization()
+        elif title == "Grant":
+            self.metadataset.add_grant()
```

Comment:
Right now property `metadataset` is of type object. You can define it at the beginning of the class as MetaDataSet Type, so all the revoked function (ex. `add_person()`) is known.

Reply:
done.

I'll actually have to work a bit more on this file in terms of typing and documentation.